main.py

```diff
@@ -134,26 +279,34 @@ async def send_wrs(self, scores_json, saved_state):
                                 saved_state["recent_scores"].remove(last_score)
                                 break
 
-                            msg = await channel.fetch_message(last_score["message_id"])
+                            for channel in channels:
+                                for message in last_score["messages"]:
+                                    if channel.id == message["channel_id"]:
+                                        msg = await channel.fetch_message(message["message_id"])
 
-                            new_content = msg.content + "\n" + score_text
+                                        new_content = msg.content + "\n" + score_text
 
-                            log(f"Appending '{score_text}' to message {msg.id}")
-                            await msg.edit(content=new_content)
+                                        log(f"Appending '{score_text}' to message {msg.id}")
+                                        await msg.edit(content=new_content)
 
-                            edited = True
+                                        messages.append({"channel_id": channel.id, "message_id": msg.id})
+                                        edited = True
+                                        break
                             break
                     
                     if not edited:
+                        messages = []
                         # send new message
-                        msg = await channel.send(score_text)
-                        saved_state["recent_scores"].append({**score, "message_id": msg.id})
+                        for channel in channels:
+                            msg = await channel.send(score_text)
+                            messages.append({"channel_id": channel.id, "message_id": msg.id})
+                        saved_state["recent_scores"].append({**score, "messages": messages})
                     
                     if rank == 1:
-                        saved_state["video_queue"].append({**score, "message_id": msg.id})
+                        saved_state["video_queue"].append({**score, "messages": messages})
```

Comment:
What happens when it edits a message in 1 channel, but doesn't find one to edit in another?
(e.g. if a channel subscribes to updates while another one is already receiving them)

With this code it would skip updates in the channel that newly subscribed.